feature_tests/steps/event_steps.py

```diff
@@ -0,0 +1,35 @@
+from behave import then
+
+from feature_tests.steps.common import (
+    assert_type_matches,
+    parse_value,
+    read_value_from_path,
+)
+
+
+@then("the following events were raised")
+def step_impl(context):
+    i = 0
+    for row in context.table:
+        assert i < len(context.events), "event count mismatch: too few raised"
+        actual = context.events[i]
+        expected = row["event"]
+        assert (
+            type(actual).__name__ == expected
+        ), "Event expected {expected} got {actual}"
+        i = i + 1
+    assert len(context.events) == i, "event count mismatch: too many raised"
```

Comment:
pretty sure this can be compacted to:

```suggestion
    assert len(context.events) == len(context.table.rows), "event count mismatch: too many raised"
    
    for row, event in zip(context.table, context.events):
        expected = row["event"]
        assert (
            type(event).__name__ == expected
        ), "Event expected {expected} got {event}"
```